### ISSUE TITLE
Files immediately purged from cache after upload to Google Drive

### ISSUE THREAD

**Issue by thebnich** [rclone/rclone#3713]
To reproduce:
```
head -c 1G < /dev/urandom > testfile
rsync --progress testfile downloads.remote/
```

While the upload is in progress, `.cache/rclone/cache-backend/downloads/.testfile.KvG6hH` is present in the cache, and increases up to 1G as the file is written as expected. Once the upload is complete, however, the entry is gone.

Log:
```
2019/11/09 04:13:16 DEBUG : &{.testfile.KvG6hH (w)}: >Write: written=131072, err=<nil>
2019/11/09 04:13:16 DEBUG : &{.testfile.KvG6hH (w)}: Write: len=131072, offset=1073610752
2019/11/09 04:13:16 DEBUG : &{.testfile.KvG6hH (w)}: >Write: written=131072, err=<nil>
2019/11/09 04:13:16 DEBUG : &{.testfile.KvG6hH (w)}: Flush:
2019/11/09 04:13:18 DEBUG : .testfile.KvG6hH: put: uploaded to remote fs and saved in cache
2019/11/09 04:13:19 DEBUG : .testfile.KvG6hH: put: added to cache
2019/11/09 04:13:19 DEBUG : : cache: expired
2019/11/09 04:13:19 INFO  : : put: cache expired
2019/11/09 04:13:19 DEBUG : .testfile.KvG6hH: Size and modification time the same (differ by -233.53µs, within tolerance 1ms)
2019/11/09 04:13:19 DEBUG : &{.testfile.KvG6hH (w)}: >Flush: err=<nil>
2019/11/09 04:13:19 DEBUG : &{.testfile.KvG6hH (w)}: Release:
2019/11/09 04:13:19 DEBUG : .testfile.KvG6hH: WriteFileHandle.Release nothing to do
2019/11/09 04:13:19 DEBUG : &{.testfile.KvG6hH (w)}: >Release: err=<nil>
2019/11/09 04:13:19 DEBUG : .testfile.KvG6hH: Attr:
2019/11/09 04:13:19 DEBUG : .testfile.KvG6hH: >Attr: a=valid=1s ino=0 size=1073741824 mode=-rw-r--r--, err=<nil>
2019/11/09 04:13:19 DEBUG : /: Lookup: name="testfile"
2019/11/09 04:13:19 DEBUG : /: >Lookup: node=<nil>, err=no such file or directory
2019/11/09 04:13:19 DEBUG : /: Rename: oldName=".testfile.KvG6hH", newName="testfile", newDir=/
2019/11/09 04:13:19 DEBUG : Cache remote downloads:: new object 'testfile'
2019/11/09 04:13:19 DEBUG : testfile: find: error: couldn't find object (testfile)
2019/11/09 04:13:19 DEBUG : find failed: not found in either local or remote fs
2019/11/09 04:13:19 DEBUG : Cache remote downloads:: moving obj '.testfile.KvG6hH' -> testfile
2019/11/09 04:13:19 DEBUG : testfile: move: file moved
2019/11/09 04:13:19 DEBUG : .testfile.KvG6hH: move: removed from cache
2019/11/09 04:13:19 DEBUG : : cache: expired
2019/11/09 04:13:19 INFO  : : move: cache expired
2019/11/09 04:13:20 DEBUG : testfile: move: added to cache
2019/11/09 04:13:20 DEBUG : : cache: expired
2019/11/09 04:13:20 INFO  : : move: cache expired
2019/11/09 04:13:20 INFO  : .testfile.KvG6hH: Moved (server side)
2019/11/09 04:13:20 DEBUG : .testfile.KvG6hH: Updating file with testfile 0xc000653180
2019/11/09 04:13:20 DEBUG : /: >Rename: err=<nil>
```

rclone config:
```
[google-drive]
type = drive
client_id = XXX
client_secret = XXX
scope = drive
token = XXX
team_drive = XXX

[downloads]
type = cache
remote = google-drive:Downloads
chunk_size = 50M
info_age = 1y
chunk_total_size = 500G
writes = true
workers = 8
```

System:
* rclone version v1.50.1
* Ubuntu 18.04 LTS x64
* Google Drive remote

**Comment by ncw** [comment]
I suspect it is the rename doing this when the file gets renamed from `.testfile.KvG6hH` to `testfile`

What happens if you do the transfer with `cp` instead of `rsync` does it work?

**Comment by thebnich** [comment]
Aha! I assumed rclone handled renames automatically, but it looks like I need the `--track-renames` flag. Thanks!

**Comment by ncw** [comment]
`--track-renames` will work with `rclone sync` but not with `rclone mount`.  It might be worth having a go with [the latest beta](https://beta.rclone.org) (in 15-30 mins from now) as I just merged a potential fix for your issue! See #3688
